packages/flutter/lib/src/widgets/two_dimensional_viewport.dart

```diff
@@ -0,0 +1,1328 @@
+    assert(_children.containsValue(child));
+    return child.parentData! as TwoDimensionalViewportParentData;
+  }
+
+  /// Returns the active child located at the provided [ChildVicinity], if there
+  /// is one.
+  ///
+  /// This can be used by subclasses to access currently active children to make
+  /// use of their size or [TwoDimensionalViewportParentData], such as when
+  /// overriding the [paint] method.
+  ///
+  /// Returns null if there is no active child for the given [ChildVicinity].
+  RenderBox? getChildFor(ChildVicinity vicinity) => _children[vicinity];
+
+  @override
+  void attach(PipelineOwner owner) {
+    super.attach(owner);
+    _horizontalOffset.addListener(markNeedsLayout);
+    _verticalOffset.addListener(markNeedsLayout);
+    _delegate.addListener(_handleDelegateNotification);
+    for (final RenderBox child in _children.values) {
+      child.attach(owner);
+    }
+  }
+
+  @override
+  void detach() {
+    super.detach();
+    _horizontalOffset.removeListener(markNeedsLayout);
+    _verticalOffset.removeListener(markNeedsLayout);
+    _delegate.removeListener(_handleDelegateNotification);
+    for (final RenderBox child in _children.values) {
+      child.detach();
+    }
+  }
+
+  @override
+  void redepthChildren() {
+    for (final RenderBox child in _children.values) {
+      child.redepthChildren();
+    }
+  }
+
+  @override
+  void visitChildren(RenderObjectVisitor visitor) {
+    RenderBox? child = _firstChild;
+    while (child != null) {
+      visitor(child);
+      child = parentDataOf(child)._nextSibling;
+    }
+  }
+
+  @override
+  void visitChildrenForSemantics(RenderObjectVisitor visitor) {
+    // Only children that are visible should be visited, and they must be in
+    // paint order.
+    RenderBox? child = _firstChild;
+    while (child != null) {
+      final TwoDimensionalViewportParentData childParentData = parentDataOf(child);
+      if (childParentData.isVisible) {
+        visitor(child);
+      }
+      child = childParentData._nextSibling;
+    }
+  }
+
+  @override
+  List<DiagnosticsNode> debugDescribeChildren() {
+    final List<DiagnosticsNode> debugChildren = <DiagnosticsNode>[
+      ..._children.keys.map<DiagnosticsNode>((ChildVicinity index) {
+        return _children[index]!.toDiagnosticsNode(name: index.toString());
+      })
+    ];
+    return debugChildren;
+  }
+
+  @override
+  Size computeDryLayout(BoxConstraints constraints) {
+    assert(debugCheckHasBoundedAxis(Axis.vertical, constraints));
+    assert(debugCheckHasBoundedAxis(Axis.horizontal, constraints));
+    return constraints.biggest;
+  }
+
+  @override
+  bool hitTestChildren(BoxHitTestResult result, {required Offset position}) {
+    for (final RenderBox child in _children.values) {
+      final TwoDimensionalViewportParentData childParentData = parentDataOf(child);
+      final Rect childRect = childParentData.paintOffset & child.size;
+      if (childRect.contains(position)) {
```

Comment:
Doesn't `child.hitTest` below already contain this check? 

https://github.com/flutter/flutter/blob/65dfb555c0483cf2bfe5e3c35291c88634185117/packages/flutter/lib/src/rendering/box.dart#L2478

Reply:
Excellent point, fixed!

Reply:
Looks like this wasn't resolved after all?

Reply:
Oh right. I changed it, and then had a bug and ended up writing it back in. Let me double check. Thanks for pointing it out! I did not realize I ended up reverting it.